mkosi

```diff
@@ -2819,34 +2829,38 @@ def build_stuff(args):
     # always the same
     args.machine_id = uuid.uuid4().hex
 
-    cache = setup_cache(args)
+    package_cache = setup_package_cache(args)
     workspace = setup_workspace(args)
 
     # If caching is requested, then make sure we have cache images around we can make use of
     if need_cache_images(args):
 
-        # Generate the cache version of the build image, and store it as "cache-pre-dev"
-        raw, tar, root_hash = build_image(args, workspace, run_build_script=True, for_cache=True)
-        save_cache(args,
-                   workspace.name,
-                   raw.name if raw is not None else None,
-                   args.cache_pre_dev)
+        # There is no point generating a pre-dev cache image if no build script is provided
+        if args.build_script:
+            # Generate the cache version of the build image, and store it as "cache-pre-dev"
+            raw, tar, root_hash = build_image(args, workspace, run_build_script=True, for_cache=True)
+            save_cache(args,
+                       workspace.name,
+                       raw.name if raw is not None else None,
+                       args.cache_pre_dev)
 
-        remove_artifacts(args, workspace.name, raw, tar, run_build_script=True)
+            remove_artifacts(args, workspace.name, raw, tar, run_build_script=True)
 
         # Generate the cache version of the build image, and store it as "cache-pre-inst"
         raw, tar, root_hash = build_image(args, workspace, run_build_script=False, for_cache=True)
-        save_cache(args,
-                   workspace.name,
-                   raw.name if raw is not None else None,
-                   args.cache_pre_inst)
-        remove_artifacts(args, workspace.name, raw, tar, run_build_script=False)
-
-    # Run the image builder for the first (develpoment) stage in preparation for the build script
-    raw, tar, root_hash = build_image(args, workspace, run_build_script=True)
-
-    run_build_script(args, workspace.name, raw)
-    remove_artifacts(args, workspace.name, raw, tar, run_build_script=True)
+        if raw:
+            save_cache(args,
+                       workspace.name,
+                       raw.name if raw is not None else None,
```

Comment:
`raw` was already determined to be truthy here, so you could just use `raw.name`.